include/SampleFrame.h

```diff
@@ -0,0 +1,234 @@
+/*
+ * SampleFrame.h - Representation of a stereo sample
+ *
+ * Copyright (c) 2004-2009 Tobias Doerffel <tobydox/at/users.sourceforge.net>
+ * Copyright (c) 2024- Michael Gregorius
+ *
+ * This file is part of LMMS - https://lmms.io
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program (see COPYING); if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA.
+ *
+ */
+
+#pragma once
+
+#ifndef SAMPLEFRAME_H
+#define SAMPLEFRAME_H
+
+#include "lmms_basics.h"
+
+#include <array>
+
+#include <cmath>
+#include <algorithm>
+
+
+namespace lmms
+{
+
+class SampleFrame
+{
+public:
+	SampleFrame() : SampleFrame(0., 0.)
+	{
+	}
+
+	SampleFrame(sample_t value) : SampleFrame(value, value)
+	{
+	}
+
+	SampleFrame(sample_t left, sample_t right) :
+		m_samples({ left, right })
+	{
+	}
+
+	sample_t* data()
+	{
+		return m_samples.data();
+	}
+
+	const sample_t* data() const
+	{
+		return m_samples.data();
+	}
+
+	sample_t& left()
+	{
+		return m_samples[0];
+	}
+
+	const sample_t& left() const
+	{
+		return m_samples[0];
+	}
+
+	void setLeft(const sample_t& value)
+	{
+		m_samples[0] = value;
+	}
+
+	sample_t& right()
+	{
+		return m_samples[1];
+	}
+
+	const sample_t& right() const
+	{
+		return m_samples[1];
+	}
+
+	void setRight(const sample_t& value)
+	{
+		m_samples[1] = value;
+	}
+
+	sample_t& operator[](size_t index)
+	{
+		return m_samples[index];
+	}
+
+	const sample_t& operator[](size_t index) const
+	{
+		return m_samples[index];
+	}
+
+	SampleFrame& operator=(float v)
+	{
+		left() = v;
+		right() = v;
+
+		return *this;
+	}
```

Comment:
The same goes for this as for the implicit constructor taking `sample_t`. If that constructor is made explicit, I think it makes sense to remove this assignment operator, instead converting the sample to a sample frame and assigning from that.

Reply:
Done with commit 4878d1ca.